operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventFilters.java

```diff
@@ -0,0 +1,163 @@
+package io.javaoperatorsdk.operator.processing.event.internal;
+
+import io.fabric8.kubernetes.client.CustomResource;
+
+/**
+ * Convenience implementations of, and utility methods for, {@link CustomResourceEventFilter}.
+ */
+public final class CustomResourceEventFilters {
+
+  private static final CustomResourceEventFilter<CustomResource> USE_FINALIZER =
+      (configuration, oldResource, newResource) -> {
+        if (configuration.useFinalizer()) {
+          final var finalizer = configuration.getFinalizer();
+          boolean oldFinalizer = oldResource == null || oldResource.hasFinalizer(finalizer);
```

Comment:
It would be good to write unit tests for these methods, since there is non-trivial logic.